### DISCUSSION TITLE
Would benefit from comprehensive testsuite

### DISCUSSION THREAD

**Discussion by rchl** [vuejs/language-tools#2239] [General]
Volar is quite a complex piece of software. Its own code is complex but also it has to interact with many different external components (typescript, different and constantly changing versions of Vue, style parsers, intellisense servics, eslint and what not). It's very easy for things to break and regress (and it happens quite often in fact). I'm surprised that such a huge project with so many users doesn't really have comprehensive testsuite. There are some tests but doesn't really seem like much. And also new tests should be added whenever something is fixed to ensure that it doesn't regress again.

Do you have any concrete plans on addressing this in the future @johnsoncodehk? Is there something that people could do to help (not necessarily me as I don't really have time to take on another OS project)?

(this in fact could be converted into a discussion)

**Comment by johnsoncodehk** [comment]
At present, I need to prioritize clearing the backlog and then I have to preparing for interviews. It may be difficult to spare time for maintenance in the future, so in the limited time, improving the test is not the top priority. If I can work on this full-time for a long time in the future, I hope to review all past bugs and improve testing.

Also now you can add your repo to https://github.com/volarjs/ecosystem-ci, I will test it before each time release, this ensures that new releases at least don't accidentally break tests of your project .
